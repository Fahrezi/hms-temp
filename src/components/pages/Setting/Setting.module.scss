.settings {
  & {
    padding: 2rem;
  }

  &__title {
    font-size: 2rem;
    font-weight: 600;
    margin-bottom: 2rem;
  }

  &__content {
    max-width: 600px;
  }

  &__section {
    background: white;
    border-radius: 0.5rem;
    padding: 1.5rem;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
    margin-bottom: 1.5rem;

    &--heading {
      font-size: 1.25rem;
      font-weight: 500;
      margin-bottom: 1rem;
    }
  }

  &__option {
    margin-bottom: 1rem;

    &--label {
      display: flex;
      align-items: center;
      cursor: pointer;

      &--checkbox {
        margin-right: 0.75rem;
      }

      &--text {
        color: #333;
      }
    }
  }
}
