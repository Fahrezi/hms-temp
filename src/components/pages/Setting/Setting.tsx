import styles from './Setting.module.scss';

const Setting = () => {
  return (
    <div className={styles.settings}>
      <h1 className={styles.settings__title}>Settings</h1>
      <div className={styles.settings__content}>
        <div className={styles.settings__section}>
          <h2 className={styles['settings__section--heading']}>Preferences</h2>
          <div className={styles.settings__option}>
            <label className={styles['settings__option--label']}>
              <input type='checkbox' className={styles['settings__option--label--checkbox']} />
              <span className={styles['settings__option--label--text']}>Enable notifications</span>
            </label>
          </div>
          <div className={styles.settings__option}>
            <label className={styles['settings__option--label']}>
              <input type='checkbox' className={styles['settings__option--label--checkbox']} />
              <span className={styles['settings__option--label--text']}>Dark mode</span>
            </label>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Setting;
