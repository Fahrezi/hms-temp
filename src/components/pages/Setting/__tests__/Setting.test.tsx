import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Setting from '@/components/pages/Setting';

describe('Setting Component', () => {
  it('renders the Setting component correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Setting />
      </MemoryRouter>,
    );

    expect(screen.getByRole('heading', { level: 1, name: 'Settings' })).toBeInTheDocument();
    expect(screen.getByRole('heading', { level: 2, name: 'Preferences' })).toBeInTheDocument();

    expect(screen.getByLabelText('Enable notifications')).toBeInTheDocument();
    expect(screen.getByLabelText('Dark mode')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('allows toggling the Enable notifications checkbox', () => {
    render(
      <MemoryRouter>
        <Setting />
      </MemoryRouter>,
    );

    const notificationsCheckbox = screen.getByLabelText('Enable notifications');
    expect(notificationsCheckbox).not.toBeChecked();

    fireEvent.click(notificationsCheckbox);
    expect(notificationsCheckbox).toBeChecked();

    fireEvent.click(notificationsCheckbox);
    expect(notificationsCheckbox).not.toBeChecked();
  });

  it('allows toggling the Dark mode checkbox', () => {
    render(
      <MemoryRouter>
        <Setting />
      </MemoryRouter>,
    );

    const darkModeCheckbox = screen.getByLabelText('Dark mode');
    expect(darkModeCheckbox).not.toBeChecked();

    fireEvent.click(darkModeCheckbox);
    expect(darkModeCheckbox).toBeChecked();

    fireEvent.click(darkModeCheckbox);
    expect(darkModeCheckbox).not.toBeChecked();
  });
});
