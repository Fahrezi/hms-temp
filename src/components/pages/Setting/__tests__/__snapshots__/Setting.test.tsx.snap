// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Setting Component > renders the Setting component correctly 1`] = `
<div>
  <div
    class="_settings_7e34ff"
  >
    <h1
      class="_settings__title_7e34ff"
    >
      Settings
    </h1>
    <div
      class="_settings__content_7e34ff"
    >
      <div
        class="_settings__section_7e34ff"
      >
        <h2
          class="_settings__section--heading_7e34ff"
        >
          Preferences
        </h2>
        <div
          class="_settings__option_7e34ff"
        >
          <label
            class="_settings__option--label_7e34ff"
          >
            <input
              class="_settings__option--label--checkbox_7e34ff"
              type="checkbox"
            />
            <span
              class="_settings__option--label--text_7e34ff"
            >
              Enable notifications
            </span>
          </label>
        </div>
        <div
          class="_settings__option_7e34ff"
        >
          <label
            class="_settings__option--label_7e34ff"
          >
            <input
              class="_settings__option--label--checkbox_7e34ff"
              type="checkbox"
            />
            <span
              class="_settings__option--label--text_7e34ff"
            >
              Dark mode
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
